src/BigBlueButton.php

```diff
@@ -443,14 +489,59 @@ public function setJSessionId($jSessionId)
         $this->jSessionId = $jSessionId;
     }
 
+    private function isJson($string)
+    {
+        return is_string($string) && is_array(json_decode($string, true)) && (json_last_error() == JSON_ERROR_NONE) ? true : false;
+    }
+
     /* ____________________ INTERNAL CLASS METHODS ___________________ */
 
+    /**
+     * A private utility method used by other public methods to process JSON responses.
+     *
+     * @param string $url
+     *
+     * @return string
+     * @throws \RuntimeException
+     */
+    private function processJsonResponse($url)
+    {
+        if (extension_loaded('curl')) {
+            $ch = curl_init();
+            if (!$ch) {
+                throw new \RuntimeException('Unhandled curl error: ' . curl_error($ch));
+            }
+
+            curl_setopt($ch, CURLOPT_SSL_VERIFYPEER, 1);
+            curl_setopt($ch, CURLOPT_ENCODING, 'UTF-8');
+            curl_setopt($ch, CURLOPT_URL, $url);
+            curl_setopt($ch, CURLOPT_RETURNTRANSFER, 1);
+            curl_setopt($ch, CURLOPT_FOLLOWLOCATION, 1);
+            curl_setopt($ch, CURLOPT_CONNECTTIMEOUT, 10);
+
+            $data = curl_exec($ch);
+            if ($data === false) {
+                throw new \RuntimeException('Unhandled curl error: ' . curl_error($ch));
+            }
+            curl_close($ch);
+
+            return $data;
```

Comment:
I think we have the same in `processXMLResponse`. Can we abstract the curl stuff and stay DRY?